src/ore/src/task.rs

```diff
@@ -0,0 +1,182 @@
+#[track_caller]
+pub fn spawn<Fut, Name, NameClosure>(_nc: NameClosure, future: Fut) -> JoinHandle<Fut::Output>
+where
+    Name: AsRef<str>,
+    NameClosure: FnOnce() -> Name,
+    Fut: Future + Send + 'static,
+    Fut::Output: Send + 'static,
+{
+    tokio::spawn(future)
+}
+
+/// Spawns a task on the executor.
+///
+/// See [`spawn`](tokio::spawn) for
+/// more details.
+#[cfg(all(tokio_unstable, feature = "task"))]
+#[track_caller]
+pub fn spawn<Fut, Name, NameClosure>(nc: NameClosure, future: Fut) -> JoinHandle<Fut::Output>
+where
+    Name: AsRef<str>,
+    NameClosure: FnOnce() -> Name,
+    Fut: Future + Send + 'static,
+    Fut::Output: Send + 'static,
+{
+    task::Builder::new().name(nc().as_ref()).spawn(future)
+}
+
+/// Spawns blocking code on the blocking threadpool.
+///
+/// See [`spawn_blocking`](::tokio::task::spawn_blocking)
+/// for more details.
+#[cfg(not(all(tokio_unstable, feature = "task")))]
+#[track_caller]
+pub fn spawn_blocking<Function, Output, Name, NameClosure>(
+    _nc: NameClosure,
+    function: Function,
+) -> JoinHandle<Output>
+where
+    Name: AsRef<str>,
+    NameClosure: FnOnce() -> Name,
+    Function: FnOnce() -> Output + Send + 'static,
+    Output: Send + 'static,
+{
+    task::spawn_blocking(function)
+}
+
+/// Spawns blocking code on the blocking threadpool.
+///
+/// See [`spawn_blocking`](::tokio::task::spawn_blocking)
+/// for more details.
+#[cfg(all(tokio_unstable, feature = "task"))]
+#[track_caller]
+pub fn spawn_blocking<Function, Output, Name, NameClosure>(
+    nc: NameClosure,
+    function: Function,
+) -> JoinHandle<Output>
+where
+    Name: AsRef<str>,
+    NameClosure: FnOnce() -> Name,
+    Function: FnOnce() -> Output + Send + 'static,
+    Output: Send + 'static,
+{
+    task::Builder::new()
+        .name(nc().as_ref())
+        .spawn_blocking(function)
+}
+
+pub trait RuntimeExt {
+    #[track_caller]
+    fn spawn_blocking_named<Function, Output, Name, NameClosure>(
+        &self,
+        nc: NameClosure,
+        function: Function,
+    ) -> JoinHandle<Output>
+    where
+        Name: AsRef<str>,
+        NameClosure: FnOnce() -> Name,
+        Function: FnOnce() -> Output + Send + 'static,
+        Output: Send + 'static;
+
+    #[track_caller]
+    fn spawn_named<Fut, Name, NameClosure>(
+        &self,
+        _nc: NameClosure,
+        future: Fut,
+    ) -> JoinHandle<Fut::Output>
+    where
+        Name: AsRef<str>,
+        NameClosure: FnOnce() -> Name,
+        Fut: Future + Send + 'static,
+        Fut::Output: Send + 'static;
+}
+
+impl RuntimeExt for Arc<Runtime> {
+    fn spawn_blocking_named<Function, Output, Name, NameClosure>(
+        &self,
+        nc: NameClosure,
+        function: Function,
+    ) -> JoinHandle<Output>
+    where
+        Name: AsRef<str>,
+        NameClosure: FnOnce() -> Name,
+        Function: FnOnce() -> Output + Send + 'static,
+        Output: Send + 'static,
+    {
+        let _g = self.enter();
```

Comment:
ah smart! when I saw your tokio console PR go by, I looked at how we could attach names in persist and didn't think of this